aws/credentials/stscreds/assume_role_provider.go

```diff
@@ -0,0 +1,116 @@
+package stscreds
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"time"
+)
+
+// assumeRoler represents the minimal subset of the STS API used here.
+type assumeRoler interface {
```

Comment:
While technically you can create a private interface for the assumeRoler it would be better if it was public. I suggest a public `AssumeRoler` interface in this package so that it will be documented and easily discoverable to people using the STS role provider